tools/wpt/android.py

```diff
@@ -15,141 +15,266 @@
+    "emulator_avd_name": "mozemulator-x86_64",
+    "emulator_extra_args": [
+        "-skip-adb-auth",
+        "-verbose",
+        "-show-kernel",
+        "-ranchu",
+        "-selinux", "permissive",
+        "-memory", "3072",
+        "-cores", "4",
+        "-skin", "800x1280",
+        "-gpu", "on",
+        "-no-snapstorage",
+        "-no-snapshot",
+        "-no-window",
+        "-no-accel",
+        "-prop", "ro.test_harness=true"
+    ],
+    "emulator_extra_config": {
+        "hw.keyboard": "yes",
+        "hw.lcd.density": "320",
+        "disk.dataPartition.size": "4000MB",
+        "sdcard.size": "600M"
+    }
+}
+
+
+def do_delayed_imports(paths):
     global android_device
     from mozrunner.devices import android_device
+
     android_device.TOOLTOOL_PATH = os.path.join(os.path.dirname(__file__),
                                                 os.pardir,
                                                 "third_party",
                                                 "tooltool",
                                                 "tooltool.py")
+    android_device.EMULATOR_HOME_DIR = paths["emulator_home"]
 
 
 def get_parser_install():
     parser = argparse.ArgumentParser()
+    parser.add_argument("--path", dest="dest", action="store", default=None,
+                        help="Root path to use for emulator tooling")
     parser.add_argument("--reinstall", action="store_true", default=False,
                         help="Force reinstall even if the emulator already exists")
+    parser.add_argument("--prompt", action="store_true",
+                        help="Enable confirmation prompts")
+    parser.add_argument("--no-prompt", dest="prompt", action="store_false",
+                        help="Skip confirmation prompts")
     return parser
 
 
 def get_parser_start():
-    return get_parser_install()
+    parser = get_parser_install()
+    parser.add_argument("--device-serial", action="store", default=None,
+                        help="Device serial number for Android emulator, if not emulator-5554")
+    return parser
 
 
-def get_sdk_path(dest):
+def get_paths(dest):
+    os_name = platform.system().lower()
+
     if dest is None:
         # os.getcwd() doesn't include the venv path
-        dest = os.path.join(wpt_root, venv_dir())
-    dest = os.path.join(dest, 'android-sdk')
-    return os.path.abspath(os.environ.get('ANDROID_SDK_PATH', dest))
+        base_path = os.path.join(wpt_root, venv_dir(), "android")
+    else:
+        base_path = dest
+
+    paths = {
+        "base": base_path,
+        "sdk": os.environ.get("ANDROID_SDK_HOME", os.path.join(base_path, f"android-sdk-{os_name}")),
+    }
+    paths["avd"] = os.environ.get("ANDROID_AVD_HOME", os.path.join(paths["sdk"], ".android", "avd"))
+    paths["sdk_tools"] = os.path.join(paths["sdk"], "cmdline-tools", CMDLINE_TOOLS_VERSION_STRING)
+    paths["emulator_home"] = os.path.dirname(paths["avd"])
+    return paths
+
+
+def get_sdk_manager(paths):
```

Comment:
Nit: get_sdk_manager_path? get_sdk_manager seems to imply this will return a sdk_manager object.